### PR TITLE
feat(app): handle non-deterministic protocols

### PR DESCRIPTION
**Pull request by jgbowser**
# Overview

When a protocol run is non-deterministic we now hide the jump to the current step button, hide the progress meter, and only display the total count of commands in the run to avoid any confusion or misdirection.

closes RLAB-299

# Test Plan

Tested locally during development with automated tests and by short-circuiting some logic to force a "non-deterministic state"

# Changelog

- conditionally render jump to step button in `RunPreview` to only show when a run is deterministic
- Hide progress bar in `RunProgressMeter` when a protocol is non-deterministic
- Display current step as `total-run-commands/?` when a protocol is non-deterministic
- adds a test file for the `RunProgressMeter` component

# Review requests

Please give this a run on a robot, I am not able to test this in action using only the Sim.

# Risk assessment
fairly low, just UI changes for edge cases

### CODE REVIEW

File: app/src/organisms/RunProgressMeter/index.tsx [hunk 1]
```diff
@@ -54,12 +60,20 @@ export function RunProgressMeter(props: RunProgressMeterProps): JSX.Element {
   const lastRunCommandKey = useLastRunCommandKey(runId)
   const lastRunCommandIndex =
     analysisCommands.findIndex(c => c.key === lastRunCommandKey) ?? 0
+  const lastRunCommandIndexFromAllCommands =
```
Comment (b-cooper) [inline app/src/organisms/RunProgressMeter/index.tsx]: It looks like the actual value that we want is from the `meta` of `allCommandsQuery` response.  It includes a `totalLength` key whose value we should be able to sub in for the numerator in the case that the run is non-deterministic

Comment (jgbowser) [reply app/src/organisms/RunProgressMeter/index.tsx]: just to make sure I understand then, the **denominator** should be a "?" and the **numerator** should be the `totalLength` of the commands: `totalLength / ?`

File: app/src/organisms/RunProgressMeter/index.tsx [hunk 2]
```diff
@@ -54,12 +60,20 @@ export function RunProgressMeter(props: RunProgressMeterProps): JSX.Element {
...
+    runCommands.findIndex(c => c.key === lastRunCommandKey) ?? 0
+  const { data: commandDetails } = useCommandQuery(
+    runId,
+    runCommands[lastRunCommandIndexFromAllCommands]?.id
+  )
```
Comment (jgbowser) [inline app/src/organisms/RunProgressMeter/index.tsx]: @b-cooper I'm now grabbing the `totalLength` from `meta` and displaying that, but I believe I still need these lines to be able to access the command text from the run rather than the analysis as I'm currently doing in L 89-96, correct?

Comment (b-cooper) [reply app/src/organisms/RunProgressMeter/index.tsx]: yeah,  that's correct 👍

**Review by b-cooper — APPROVED** [review]
Changes look good, thanks for making those tweaks 👍

**Comment by jgbowser** [comment]
> Changes look good, thanks for making those tweaks 👍

No problem, am I good to merge this then, or is there still some dev testing required?
